### PR TITLE
feat: pgbouncer plugin

### PR DESCRIPTION
**Pull request by algchoo**
### Proposed Change

Ported PGBouncer plugin

configuration used:
```
receivers:
  plugin:
    path: ./plugins/pgbouncer_logs.yaml
    parameters:
      file_path:
        - /Users/austingeorgiades/projects/observiq-otel-collector/in
      start_at: beginning

exporters:
  logging:
    logLevel: debug

service:
  pipelines:
    logs:
      receivers:
      - plugin
      exporters:
      - logging
```

input/test data:
```
2021-05-17 16:16:40.613 1354 LOG Stats: 0 xacts/s, 0 queries/s, in 0 B/s, out 0 B/s, xact 0 us, query 0 us wait time 0 us
2021-05-18 13:12:05.011 UTC [1] LOG stats: 927 xacts/s, 935 queries/s, in 420598 B/s, out 687149 B/s, xact 893 us, query 816 us, wait 1 us
2021-05-18 13:12:05.191 UTC [1] LOG C-0x55a12ff32ad0: collectors/bindplane@10.1.0.70:56614 closing because: client close request (age=82s)
2021-05-18 12:30:05.591 12788 WARNING C-0x564043c80e20: (nodb)/(nouser)@10.33.104.53:37688 Pooler Error: No such database: bob
```

sample output:
```
LogRecord #7
ObservedTimestamp: 2022-05-16 15:39:03.880806 +0000 UTC
Timestamp: 1970-01-01 00:00:00 +0000 UTC
Severity: Warn
Body: 2021-05-18 12:30:05.591 12788 WARNING C-0x564043c80e20: (nodb)/(nouser)@10.33.104.53:37688 Pooler Error: No such database: bob
Attributes:
     -> log.file.name: STRING(in)
     -> pid: STRING(12788)
     -> severity: STRING(WARNING)
     -> message: STRING(C-0x564043c80e20: (nodb)/(nouser)@10.33.104.53:37688 Pooler Error: No such database: bob)
     -> timestamp: STRING(2021-05-18 12:30:05.591)
     -> log_type: STRING(pgbouncer)
Trace ID:
Span ID:
Flags: 0
```

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: plugins/pgbouncer_logs.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      start_at: {{ .start_at }}
+      operators:
+        - type: regex_parser
+          regex: '^(?P<timestamp>\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}.\d{3}(\s*[^\d\s]{3})?)\s*(\[)?(?P<pid>\d+)(\])? (?P<severity>\w+) (?P<message>.*)'
+          severity:
+            parse_from: attributes.severity
+            mapping:
+              critical: fatal
+              info: log
+              notice: stats
```
Comment (BinaryFissionGames) [inline plugins/pgbouncer_logs.yaml:30]: notice isn't a valid alias in opentelemetry like it was in stanza; We've been mapping it to `info2` in general, I think.

File: plugins/pgbouncer_logs.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        - '{{ $fp }}'
+        {{end}}
+      start_at: {{ .start_at }}
+      operators:
+        - type: regex_parser
+          regex: '^(?P<timestamp>\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}.\d{3}(\s*[^\d\s]{3})?)\s*(\[)?(?P<pid>\d+)(\])? (?P<severity>\w+) (?P<message>.*)'
+          severity:
+            parse_from: attributes.severity
+            mapping:
+              critical: fatal
```
Comment (BinaryFissionGames) [inline plugins/pgbouncer_logs.yaml:28]: critical also isn't a valid mapping; `fatal` severity does already exist in otel though, so we can just map directly by omitting this line, I think. 

If you're curious, here's the list of aliases recognized here in otel, it's a bit tricky 'cuz some of them port directly from stanza and other do not:
https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/types/severity.md

File: plugins/pgbouncer_logs.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+version: 0.0.1
+title: PgBouncer
+description: Log parser for PgBouncer
+parameters:
+  - name: file_path
+    type: "[]string"
+    default: "/var/log/pgbouncer/pgbouncer.log"
```
Comment (cpheps) [inline plugins/pgbouncer_logs.yaml:7]: This needs to be a yaml array. See other plugins for examples.

**Review by cpheps — CHANGES_REQUESTED** [review]

**Review by BinaryFissionGames — COMMENTED** [review]
These in addition to what Corbin said, looks like the output isn't having the timestamp or message parsed.

**Review by cpheps — COMMENTED** [review]
@algchoo Just to verify you tested against live system?

**Comment by algchoo** [comment]
> @algchoo Just to verify you tested against live system?



> @algchoo Just to verify you tested against live system?

@cpheps @BinaryFissionGames We we're initially testing against sample logs, not from an active live environment. Some changes/fixes have been made and it has been tested against a live system, successfully.

**Review by cpheps — APPROVED** [review]

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**